modules/infra/cli/notifications.c

```diff
@@ -48,6 +51,42 @@ static cmd_status_t notifications_dump(const struct gr_api_client *c, const stru
 				printf("Iface reconfigured: %s\n", p->iface.name);
 			}
 			break;
+		case IP_EVENT_ADDR_ADD:
+			if (n->payload_len == sizeof(*nh)) {
+				nh = (struct gr_nexthop *)&n[1];
+				printf("IP address add: iface[%d] " ADDR_F "\n",
+				       nh->iface_id,
+				       ADDR_W(nh->family),
+				       &nh->addr);
+			}
+			break;
+		case IP_EVENT_ADDR_DEL:
+			if (n->payload_len == sizeof(*nh)) {
```

Comment:
Why check the size here? Is there a chance that you receive a IP_EVENT_ADDR_DEL message without a struct gr_nexthop payload?